virtManager/config.py

```diff
@@ -212,6 +212,17 @@ def embeddable_graphics(self):
         ret = ["vnc", "spice"]
         return ret
 
+    def get_perfered_graphics_index(self, gdevs):
+        try:
+            for idx, dev in enumerate(gdevs):
+                if dev.type in self.embeddable_graphics():
+                    return idx
+            if len(gdevs) > 0:
+                return 0
+            return None
+        except:
+            return None
```

Comment:
This whole function, and the `embeddable_graphics` function, can be moved into console.py actually. `embeddable_graphics` is only called there, it's a remnant from earlier days. Moving 'embeddable_graphics` could be a prep patch

Reply:
Thanks very much for your suggestions! I have moved `embeddable_graphics` to console.py and removed `get_prefered_grahpics_index`  in v2  after some optimization.